File: core/src/com/mygdx/game/characters/State/RunningLeft.java
package com.mygdx.game.characters.State;

import com.badlogic.gdx.math.Vector3;
import com.mygdx.game.characters.GameCharacter;
import com.mygdx.game.core.NodeHolder;

public class RunningLeft extends State {
    public RunningLeft(GameCharacter character) {
        super(character);
    }

    public void stand() {character.state = new Stand(character);}
    public void runLeft() {
        Vector3 sub = new Vector3(-character.walkSpeed,0,0);
        sub.rotate(NodeHolder.rotAxis,character.globalRot);
        character.pos.add(sub);
    }
    public void runRight() {character.state = new Stand(character);}
    public void walkLeft() {}
    public void walkRight() {}

    @Override
    public void climbUp() {

    }

    @Override
    public void climbDown() {

    }
}

File: web/src/characters/state/RunningLeft.ts
import { State } from './State';
import { Stand } from './Stand';

export class RunningLeft extends State {
  stand(): void {
    this.character.fsm = new Stand(this.character);
  }
  runLeft(): void {
    this.character.addLogicalPos(-this.character.walkSpeed, 0);
  }
  runRight(): void {
    this.character.fsm = new Stand(this.character);
  }
  walkLeft(): void {}
  walkRight(): void {}
  climbUp(): void {}
  climbDown(): void {}
}
